homeassistant/components/intellifire/number.py

```diff
@@ -25,21 +25,66 @@ async def async_setup_entry(
     """Set up the fans."""
     coordinator: IntellifireDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
 
-    description = NumberEntityDescription(
-        key="flame_control",
-        name="Flame control",
-        icon="mdi:arrow-expand-vertical",
-    )
-
     async_add_entities(
         [
             IntellifireFlameControlEntity(
-                coordinator=coordinator, description=description
-            )
+                coordinator=coordinator,
+                description=NumberEntityDescription(
+                    key="flame_control",
+                    name="Flame control",
+                    icon="mdi:arrow-expand-vertical",
+                ),
+            ),
+            IntellifireTimerControlEntity(
+                coordinator=coordinator,
+                description=NumberEntityDescription(
+                    key="sleep_control",
+                    name="Sleep timer",
+                    icon="mdi:bed-clock",
+                ),
+            ),
         ]
     )
 
 
+@dataclass
+class IntellifireTimerControlEntity(IntellifireEntity, NumberEntity):
+    """Countdown timer control entity."""
+
+    _attr_native_max_value: float = 180
+    _attr_native_min_value: float = 0
+    _attr_native_step: float = 1
+    _attr_mode: NumberMode = NumberMode.AUTO
+    _attr_native_unit_of_measurement = "minutes"
+
+    def __init__(
+        self,
+        coordinator: IntellifireDataUpdateCoordinator,
+        description: NumberEntityDescription,
+    ) -> None:
+        """Initialize fireplace timer."""
+        super().__init__(coordinator, description)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the current Timer value in minutes."""
+        value = int(self.coordinator.read_api.data.timeremaining_s / 60)
+        return value
```

Comment:
The type hint is wrong here, and there is no need for the intermediate variable.
```suggestion
    @property
    def native_value(self) -> int:
        """Return the current Timer value in minutes."""
        return = int(self.coordinator.read_api.data.timeremaining_s / 60)
```